README.md

```diff
@@ -1 +1,139 @@
-# microcosm-fastapi
\ No newline at end of file
+# microcosm-fastapi
+
+A bridge between FastAPI and microcosm. Provide state-of-the-art speed of hosting microservices in Python along with dependency injection.
+
+## Top Features
+
+- Async compliant protocol to add new microservice functions
+- Specify API requests and responses with business-logic typehinting, while following strongly conventioned CRUD operations to access the database
+- Async postgres support using the latest SQLAlchemy 1.4 (still in beta), to support more concurrent client users with fewer CPU blocking requests
+- Automatic generation of interactive documentation, available on localhost:5000/docs when doing development work
+
+## Migration from microcosm-flask
+
+If you're using microcosm already, there's a high chance you're using `microcosm-flask` as your communication layer. This library attempts to give a relatively straightforward migration path from microcosm-flask by using the same abstraction and function names where possible. In order to truly leverage the best design decisions that went into FastAPI, however, we also need to refactor some of our logic into the new FastAPI paradigm.
+
+### Resources
+
+When defining resources in flask, you're likely used to defining a schema for everything that you'll ever need to serialize from client users. This includes `New` schemas for creating new objects, `Standard` schemas for retrieving objects from the database, and `Search` schemas for extracting the URL parameters that can search against the database layer. These are defined via `marshmallow` which provides the serialization layer to convert from json->python and python->json. Something like this:
+
+```
+from microcosm_flask.paging import PageSchema
+from marshmallow import Schema, fields
+
+class NewPizzaSchema(Schema):
+    toppings = fields.String(required=True)
+
+class PizzaSchema(NewPizzaSchema):
+    id: fields.UUID(required=True)
+
+class PizzaSearchSchema(PageSchema):
+    toppings = fields.String(required=False)
+```
+
+Instead of marshmallow, FastAPI makes extensive use of `pydantic` to provide the validation layers. Pydantic is a more modern library in comparison. It uses python typehints in order to define expected field types and has more built-in functionality when compared to marshmallow. It's straightforward to convert the above definitions into ones that are pydantic compatible. Note that we remove the `PizzaSearchSchema` entirely because this definition will be specified in another file.
```

Comment:
You can automate a lot of this see: https://pydantic-docs.helpmanual.io/datamodel_code_generator/
Here is an example of taking the `ariza` swagger and the outputted pydantic schemas:
https://github.com/globality-corp/globality-scratchpad/blob/master/rob.hand/fastapi_example/ariza-model.py

Gets you probably 50-70% of the way there as it understands all the basic fields as described by our swagger defs. The main additional elements are it dupes the existing enums